runtime/include/sandbox_setup_arguments.h

```diff
@@ -15,12 +17,13 @@ sandbox_setup_arguments(struct sandbox *sandbox)
 {
 	assert(sandbox != NULL);
 	int32_t argument_count = 0;
-	/* whatever gregor has, to be able to pass arguments to a module! */
-	sandbox->arguments_offset = local_sandbox_context_cache.memory.size;
-	assert(local_sandbox_context_cache.memory.start == sandbox->memory.start);
-	expand_memory();
 
-	int32_t string_off = sandbox->arguments_offset;
+	/* Copy arguments into linear memory. It seems like malloc would clobber this, but I think this goes away in
+	 * WASI, so not worth fixing*/
+
+	sandbox->arguments_offset = wasm_memory_get_size(sandbox->memory);
+	int rc                    = wasm_memory_expand(sandbox->memory, WASM_PAGE_SIZE);
```

Comment:
I do not have a good understanding of wasmception libc init logic. It seemed like reading the initial and max page size from the module caused segfaults. I believe I padded this by a page to get this working. This is likely a hack, but I didn't see much sense in investing in learning wasmception internals at this point.